includes/Checker/Checks/Plugin_Readme_Check.php

```diff
@@ -198,12 +198,69 @@ private function check_stable_tag( Check_Result $result, string $readme_file, Pa
 	private function check_for_warnings( Check_Result $result, string $readme_file, Parser $parser ) {
 		$warnings = $parser->warnings ? $parser->warnings : array();
 
+		// This should be ERROR rather than WARNING. So ignoring here to handle separately.
+		if ( isset( $warnings['invalid_plugin_name_header'] ) ) {
+			unset( $warnings['invalid_plugin_name_header'] );
+		}
+
 		$warning_keys = array_keys( $warnings );
 
+		$latest_wordpress_version = $this->get_wordpress_stable_version();
+
 		$ignored_warnings = array(
 			'contributor_ignored',
 		);
 
+		$messages = array(
+			'contributor_ignored'          => sprintf(
+				/* translators: %s: plugin header tag */
+				__( 'One or more contributors listed were ignored. The %s field should only contain WordPress.org usernames. Remember that usernames are case-sensitive.', 'plugin-check' ),
+				"'Contributors'"
+			),
+			'requires_php_header_ignored'  => sprintf(
+				/* translators: 1: plugin header tag; 2: Example version 5.2.4. 3: Example version 7.0. */
+				__( 'The %1$s field was ignored. This field should only contain a PHP version such as %2$s or %3$s.', 'plugin-check' ),
+				"'Requires PHP'",
+				"'5.2.4'",
+				"'7.0'"
+			),
+			'tested_header_ignored'        => sprintf(
+				/* translators: 1: plugin header tag; 2: Example version 5.0. 3: Example version 5.1. */
+				__( 'The %1$s field was ignored. This field should only contain a valid WordPress version such as %2$s or %3$s.', 'plugin-check' ),
+				"'Tested up to'",
+				"'" . number_format( floatval( $latest_wordpress_version ), 1 ) . "'",
```

Comment:
Nit: here and elsewhere I would use `(float)` instead of `floatval()`.

Even better, just cast it to a float once directly inside `get_wordpress_stable_version()` before returning the version. Then you don't need the 4 type casts here.